libsolidity/ast/Types.h

```diff
@@ -372,6 +373,14 @@ class Type
 	/// are returned without modification.
 	virtual TypeResult interfaceType(bool /*_inLibrary*/) const { return nullptr; }
 
+	/// @returns list of direct subtypes (base type for array, members for structs,
+	/// value type for mappings, empty list for ordinary types)
+	virtual std::vector<Type const*> subtypes() const { return {}; }
+	/// @returns list of direct or indirect subtypes
+	std::vector<Type const*> allSubtypes() const;
```

Comment:
Please say if the types guaranteed to be in any specific order and if the vector can contain duplicates. Also, the fact that it contains the type itself should be mentioned.

Reply:
See https://github.com/ethereum/solidity/pull/9746#discussion_r487138562.